README.md

```diff
@@ -0,0 +1,42 @@
+# Introduction
+The 'um-update_toolbox_armhf' is a 'squashfs' minimal root filesystem that can be mounted in early boot stages, with the purpose of providing tools to be able to perform complex setup and update operations. 
+
+This toolbox image is based on Linux Alpine, a security-oriented, lightweight Linux distribution based on musl libc and busybox. 
+
+On top of the standard distro we install the required tooling, such as; e2fsprogs-extra f2fs-tools rsync, resulting in an image with a compressed size of less then 2.6M at time of writing. 
+
+## Repository structure
+Root:
+- License and readme files
+- Docker files used by CI for generating an isolated build environment see...
+- Scripts for building the software
+- Configuration files for Gitlab CI
+
+Scripts folder:
+
+Update or setup scripts, the main entrypoint here is startup.sh. These 
+scripts will be placed in the toolbox '/sbin' folder  
+
+Tests folder:
+- A test to check the build environment
+- A test to validate the generated toolbox functionality
+
+## How to build
+This repository is setup to use an existing docker image from our docker registry at gitlab. The image contains all the tools to be able to build the software. When docker is not installed the build script will try to build on the host machine and will probably fail because the required tools are not installed. If you necessarily wan't to build without docker, read the tests/buildenv.sh script and check what the requirements are. 
```

Comment:
read the buildenv.sh script? run the test/buildenv script first i would say as it should tell the user what is needed/missing.